src/coreclr/vm/codeman.h

```diff
@@ -478,12 +478,19 @@ typedef struct _HeapList
     size_t              maxCodeHeapSize;// Size of the entire contiguous block of memory
     size_t              reserveForJumpStubs; // Amount of memory reserved for jump stubs in this block
 
-#if defined(TARGET_AMD64)
-    BYTE        CLRPersonalityRoutine[JUMP_ALLOCATE_SIZE];                 // jump thunk to personality routine
-#elif defined(TARGET_ARM64)
-    UINT32      CLRPersonalityRoutine[JUMP_ALLOCATE_SIZE/sizeof(UINT32)];  // jump thunk to personality routine
+#if defined(TARGET_AMD64) || defined(TARGET_ARM64)
+    BYTE*               CLRPersonalityRoutine;  // jump thunk to personality routine
 #endif
 
+    TADDR GetModuleBase()
+    {
+#if defined(TARGET_AMD64) || defined(TARGET_ARM64)
+        return (TADDR)CLRPersonalityRoutine;
+ #else
+        return (TADDR)mapBase;
```

Comment:
Why not always return `mapBase` here, even on AMD64 and ARM64?

Reply:
These values are different for dynamic methods. The HostCodeHeap prefixes all allocations storage by the TrackAllocation instance, so the CLRPersonalityRoutine is stored at an offset from the mapBase.
The ExecutionManager::GetCLRPersonalityRoutineValue() needs to return the same offset for both cases so that we don't need to check whether we are getting it for dynamic method or regular code (whether it came from the HostCodeHeap or LoaderHeap).